contracts/BlockRelayProxy.sol

```diff
@@ -12,15 +12,26 @@ import "./BlockRelayInterface.sol";
  */
 contract BlockRelayProxy {
 
+  // Address of the current controller
   address public blockRelayAddress;
+  // Current interface to the controller
   BlockRelayInterface blockRelayInstance;
 
+  // Mapping of epoch for which we can start quering to BR Controller
+  mapping(uint256 => address) public epochBr;
```

Comment:
Could this be just an array, with the same structure as `lastEpochs`?

```suggestion
  address[] public epochBr;
```

So instead of doing `epochBr[lastEpochs[0]]` you can just do `epochBr[0]`.

In other programming languages you could declare an array of `(epoch, block_relay_address)`, which is the same as having two separate arrays but less error prone.